### DISCUSSION TITLE
Suppress output of property value text

### DISCUSSION THREAD

**Discussion by lawsonmcw** [dotnet/docfx#9655] [Q&A] [answered]
My code is pretty well documented already with xml `<summary>` tags, but I'm finding if the `<value>` tag isn't populated for properties, the API documentation just repeats the class summary which is a bit of a non-sequitur.  For example:

![image](https://github.com/dotnet/docfx/assets/1609488/0e3f7984-48e7-4ace-80d4-c8b57b664ef9)

But when it comes to methods with only a xml `summary` tag it doesn't do this, which is perfect.

![image](https://github.com/dotnet/docfx/assets/1609488/9b840aef-27c2-46f1-8dba-8160c234be24)

Is there a way to suppress the output for properties with a missing `<value>` tag? I'd rather not add these to the hundreds of properties in my codebase that essentially repeats information I've already written in `<summary>`.  I've tried editing the class.tmpl.partial file in my custom template with no luck.  Is there a directive I can add to the stylesheet to suppress this output for properties? Thanks for any and all suggestions!

**Comment by BattlefieldDuck** [comment]
Could you please share the XML documentation for the function shown in the screenshot you provided?

**Comment by lawsonmcw** [comment]
Of course! Here's an excerpt of the code. XML documentation was done in Visual Studio.

```c#
namespace Balsam
{
    /// <summary>
    /// Encapsulates information about a datasource, typically from a call to a BarServer.
    /// </summary>
    public class DataSource : ICloneable
    {
        /// <summary>
        /// Gets the type of BarServer used.
        /// </summary>
        public string BarServer { get; set; }
        
        /// <summary>
        /// Loads a json text file representation.
        /// </summary>
        /// <param name="fileName"></param>
        /// <returns></returns>
        public static DataSource Load(string fileName)
        {
            var json = File.ReadAllText(fileName);
            return JsonConvert.DeserializeObject<DataSource>(json);
        }
    }
}
```
I'd love for the Property Value output to just report the type `string` similar to how the `Load` method documentation reports `DataSource` without any other commentary.

But I did just discover in writing this response if I put in an empty `<value>` tag for the BarServer property it suppresses the message carried over from the class `<summary>`.

```c#
 /// <summary>
 /// Gets the type of BarServer used.
 /// </summary>
 /// <value></value>
 public string BarServer { get; set; }
```
It's not ideal since Visual Studio seems to rely primarily on the `<summary>` tag for IntelliSense support. It would be nice to have a css setting similar to how you helped me get rid of the Inherited Members section.  But it's also not the end of the world if I have to add empty `<value>` tags to all my properties to get the API output I want.

Thank you @BattlefieldDuck! Your help has gotten me very close to a documentation site I can be proud of.

**Comment by BattlefieldDuck** [comment]
1. **Create a CSS file** 

Navigate to the `docs/template/public/main.css` directory and create a new CSS file. Add the following CSS rules to this file:

```css
.parameters dd {
    display: none;
}
```

This rule will hide all `<dd>` elements within any element with the class `.parameters`.

2. **Modify the docfx.json file** 

Update the template section as follows:

```json
"template": [
  "default",
  "modern",
  "template"
],
```

After these changes, your documentation should reflect the new styling rules. Remember to rebuild your documentation for the changes to take effect.

**Comment by lawsonmcw** [comment, marked as answer]
Hi @BattlefieldDuck. I can confirm that your CSS rule does the trick. Now to back out the 200 empty `<value>` tags I added to public properties in my brute force attempt at fixing the problem. Fortunately, it's far easier to revert the changes than it was to add them!  I'm in good shape now thanks to all your help.
